rootfs/bin/do_upgrade

```diff
@@ -0,0 +1,57 @@
+#!/usr/bin/env bash
+
+POSTGRES_OLD_VERSION=$(cat "$1"/PG_VERSION)
+
+install-old-postgres() {
+  if [ "$POSTGRES_OLD_VERSION" == "9.4" ];then
+    echo 'http://dl-cdn.alpinelinux.org/alpine/v3.3/main' >> /etc/apk/repositories    
+    apk add --update --virtual .upgrade-deps postgresql=9.4.15-r0
+  fi
+  if [ "$POSTGRES_OLD_VERSION" == "9.5" ];then
+    echo 'http://dl-cdn.alpinelinux.org/alpine/v3.4/main' >> /etc/apk/repositories
+    apk add --update --virtual .upgrade-deps postgresql=9.5.13-r0
+  fi    
+  if [ "$POSTGRES_OLD_VERSION" == "9.6" ];then
+    echo 'http://dl-cdn.alpinelinux.org/alpine/v3.6/main' >> /etc/apk/repositories
+    apk add --update --virtual .upgrade-deps postgresql=9.6.10-r0
+  fi
+  if [ "$POSTGRES_OLD_VERSION" == "10" ];then
+    echo 'http://dl-cdn.alpinelinux.org/alpine/v3.8/main' >> /etc/apk/repositories
+    apk add --update --virtual .upgrade-deps postgresql=10.5-r0                             
+  fi
+}
```

Comment:
Interesting question. I was thinking about the same but since this script only updates the existing in-cluster postgresDB, that version should never be below 9.4 since that is what came with previous versions of Deis Workflow pre-v1.19.0. So if it is ever a version below 9.4 then the user must have done something very custom. It will be on the user to figure out how to do that upgrade....

Is my thinking correct @duanhongyi ?

Reply:
Wouldn't hurt to just "die" and echo something like, "We don't support upgrades from X.Y.Z" if there is an unexpected version. Especially as the script continues, as far as I can see. ;)

Reply:
Sounds good to me. @duanhongyi do you have some time to add an echo and an exit command if the version is unsupported?

Reply:
@Cryptophobia

Reply:
@duanhongyi , sorry to bother you and enjoy your vacation! We can merge this in when you are back. :+1:

Reply:
@Cryptophobia @till

Reply:
Maybe it should do things in this order:
``` 
1. echo a message 
2. timeout and wait 2-3 minutes 
3. exit 1 
```

I think it should fail but with a very clear log message. Then the user should go back to using an older `hephy/postgres` image that does not try to invoke a script to do the upgrade. :+1: 

Any thoughts @till ?